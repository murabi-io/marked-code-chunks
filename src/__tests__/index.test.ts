import { marked } from "marked";

import extension from "../index";
import { trimLines } from "./helpers";

marked.use(extension());

describe("marked-code-chunks", () => {
  describe("given a code chunk", () => {
    it("should render chunks correctly with ~~~ fences", () => {
      const md = trimLines(`
      ~~~bash {osx, cmd="/bin/bash"}
            ls .
      ~~~
    `);
      expect(marked(md)).toMatchSnapshot();
    });
  });
});
