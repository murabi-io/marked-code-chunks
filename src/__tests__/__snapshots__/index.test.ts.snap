// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`marked-code-chunks given a code chunk should render chunks correctly with ~~~ fences 1`] = `
"<pre><code data-osx=\\"true\\" data-cmd=\\"/bin/bash\\" class=\\"language-bash\\">ls .
</code></pre>
"
`;
